lib/Echidna/Config.hs

```diff
@@ -180,3 +181,34 @@ withDefaultConfig = (`runReaderT` defaultConfig)
 -- | 'withDefaultConfig' but not for transformers
 withDefaultConfig' :: Reader EConfig a -> a
 withDefaultConfig' = (`runReader` defaultConfig)
+
+data Env = Env {
+  _cfg :: EConfig,
```

Comment:
How exactly `cfg` is used from `Env`?

Reply:
It's used by the `Has` implementations below `data Env ...` https://github.com/crytic/echidna/blob/4c4734ac1bd09a70ac63cf4577f1a329b8fcf675/lib/Echidna/Config.hs#L192-L211

There are a lot of class constraints like `Has CampaignConf x`, `Has TestConf x` throughout the codebase where `x` is the environment of the `MonadReader`. 
`EConfig` satisfies all of these and is currently passed as the reader monad environment here https://github.com/crytic/echidna/blob/16834a15e55221fb81cb37fa5cdf9631a0d50cd9/src/Main.hs#L115

To be able to map codehashes to contract names I needed to pass a mapping via the reader monads environment.
`DappInfo` contains such a mapping. 
This could be retrieved via `hasLens` given a `Has DappInfo x`.

So I needed to implement `Has DappInfo x` for reader monad environment.

I didn't want to clutter `EConfig` with `DappInfo` as `EConfig` is concerned with configuration and `DappInfo` isn't configuration in my opinion.
That's why I added the `Env` type to wrap `EConfig` and `DappInfo`, and to satisfy all required `Has` constraints.

It was very ad hoc.
I don't think it's a bad idea to have a dedicated type for the monad reader environment that is nothing concrete like the config and to which new things which are needed from the environment can be added easily.